services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -550,6 +572,18 @@ getSelfDeletingMessagesInternal = \case
       TeamFeatures.getSelfDeletingMessagesStatus tid
         <&> maybe Public.defaultSelfDeletingMessagesStatus id
 
+setSelfDeletingMessagesPaymentStatusInternal ::
+  Member TeamFeatureStore r =>
+  TeamId ->
+  Public.PaymentStatusUpdate ->
+  Galley r Public.PaymentStatus
+setSelfDeletingMessagesPaymentStatusInternal tid paymentStatusUpdate =
+  liftSem $ TeamFeatures.setFeatureSelfDeletingMessagesPaymentStatus tid (Public.PaymentStatus paymentStatus)
+  where
+    paymentStatus = case paymentStatusUpdate of
+      Public.Lock -> Public.PaymentLocked
```

Comment:
Yes, as mentioned earlier, if we have just `PaymentStatus`, and not `PaymentStatusUpdate` on top of it, things will get easier all over the place...

Reply:
Fixed it